### PR TITLE
You can add a phone number to your user profile

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: app/controllers/UserProfileController.java [hunk 1]
```diff
@@ -16,14 +16,15 @@ public void setUserProfileService(UserProfileService userProfileService) {
     }
 
     public Result getUserProfile() throws Exception {
-        UserSession session = getAuthenticatedSession();
```
Comment (alxdarksage) [inline app/controllers/UserProfileController.java:19]: Authenticated only.

File: app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java [hunk 2]
```diff
@@ -1,27 +1,67 @@
...
+    public UserProfile getProfile(String email) {
+        Account account = authService.getAccount(email);
+
+        UserProfile profile = new UserProfile();
+        profile.setFirstName(removeEmpty(account.getGivenName()));
+        profile.setLastName(removeEmpty(account.getSurname()));
+        profile.setUsername(removeEmpty(account.getUsername()));
+        profile.setEmail(removeEmpty(account.getEmail()));
+        
+        String encryptedPhone = (String)account.getCustomData().get("phone");
```
Comment (alxdarksage) [inline app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java]: Make a constant out of "Phone"

File: app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java [hunk 3]
```diff
@@ -1,27 +1,67 @@
...
 
     @Override
+    public UserProfile getProfile(String email) {
+        Account account = authService.getAccount(email);
+
+        UserProfile profile = new UserProfile();
+        profile.setFirstName(removeEmpty(account.getGivenName()));
+        profile.setLastName(removeEmpty(account.getSurname()));
+        profile.setUsername(removeEmpty(account.getUsername()));
+        profile.setEmail(removeEmpty(account.getEmail()));
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java]: Are empty usernames and emails possible? We get the account from Stormpath using email, so I would assume email addresses always exist. Similarly, our signUp API provides a username, so I assume we'll always have a username.

File: app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java [hunk 4]
```diff
@@ -1,27 +1,67 @@
...
+        
+        String encryptedPhone = (String)account.getCustomData().get("phone");
+        if (encryptedPhone != null) {
+            profile.setPhone(healthCodeEncryptor.decrypt(encryptedPhone));
+        }
```
Comment (DwayneJengSage) [inline app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java:40]: If phone number isn't specified, the field will return null. If name isn't specified, the field will return empty string. Should we change phone number to empty string or removeEmpty to null to be consistent?

**Pull request merged**